BankApp/Account.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Text;
+
+namespace BankApp
+{
+    class Account
+    {
+        public string Name { get; private set; }
+        public Guid AccountNumber { get; }
+        public decimal AccountBalance { get; private set; }
+        public List<Transfer> TransferList { get; }
+
+        public Account(Guid number)
+        {
+            AccountNumber = number;
+        }
+
+        public Account(string name)
+        {
+            Name = name;
+            AccountNumber = Guid.NewGuid();
+            AccountBalance = 1000.00m;
+            TransferList = new List<Transfer>();
+        }
+
+        public void MakeTransfer(Account receiver, decimal amount, string name, string type)
+        {
+            try
+            {
+                CheckIfValidAmount(amount);
+
+                if (type == "domestic")
+                {
+                    CheckIfValidReceiver(receiver);
+                }
+            }
+
+            catch (Exception e)
+            {
+                Console.WriteLine(e.Message);
+                return;
+            }
+
+            AccountBalance -= amount;
+            Transfer transfer = new Transfer(this, receiver, amount, name, type);
+            TransferList.Add(transfer);
+
+            if (type == "domestic")
+            {
+                receiver.AccountBalance += amount;
+            }
+
+            Console.WriteLine("\nTransfer executed successfully.");
+        }
+
+        private void CheckIfValidReceiver(Account receiver)
+        {
+            if (object.ReferenceEquals(receiver, this))
+            {
+                throw new ArgumentException("Receiver account is the same");
+            }
+        }
+
+        private void CheckIfValidAmount(decimal amount)
+        {
+            if (amount <= 0)
+            {
+                throw new ArgumentOutOfRangeException("Amount of transfer must be positive");
```

Comment:
Do not use exceptions as a conditional mechanism - its error prone, makes code more unpredictable